partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprCallDynamic.kt

```diff
@@ -21,20 +21,21 @@ import org.partiql.value.PartiQLValueType
  */
 @OptIn(PartiQLValueExperimental::class, FnExperimental::class)
 internal class ExprCallDynamic(
-    private val candidates: List<Candidate>,
+    candidates: List<Candidate>,
     private val args: Array<Operator.Expr>
 ) : Operator.Expr {
 
+    private val candidateIndex = CandidateIndex.All(candidates)
```

Comment:
I say create the index in the Compiler and pass to the ExprCallDynamic. In this scenario, we're passing an arg into the constructor just to be passed to another constructor.